spec/domain/export/pdf/messages/letter_spec.rb

```diff
@@ -12,15 +12,48 @@
   let(:letter) { messages(:letter) }
   let(:recipients) { [people(:bottom_member)] }
   let(:options) { {} }
+  let(:layer) { groups(:top_layer) }
+  let(:group) { groups(:top_group) }
+  let(:list) { mailing_lists(:leaders) }
 
   subject { described_class.new(letter, recipients, options) }
+  # move list from top layer to top group
+  before { list.update!(group: group) }
 
-  it 'renders logo from settings' do
-    expect_any_instance_of(Prawn::Document).to receive(:image).with(
-      Settings.messages.pdf.logo.to_s,
-      fit: [200, 40]
-    )
-    subject.render
+  context 'heading' do
+
+    let(:letter_setting) do
+      GroupSetting.new(var: :messages_letter, picture: fixture_file_upload('images/logo.png'))
+    end
+
+    it 'renders no logo' do
+      expect_any_instance_of(Prawn::Document).to receive(:image).never
+      subject.render
+    end
+
+    it 'renders logo from layer setting' do
+      letter.update!(heading: true)
+      letter_setting.target = layer
+      letter_setting.save!
+
+      expect_any_instance_of(Prawn::Document).to receive(:image).with(
+        /\/picture\/#{letter_setting.id}\/logo\.png/,
+        fit: [200, 40]
+      )
+      subject.render
+    end
+
+    it 'renders logo from group setting' do
+      letter.update!(heading: true)
+      letter_setting.target = group
+      letter_setting.save!
+
+      expect_any_instance_of(Prawn::Document).to receive(:image).with(
+        /\/picture\/#{letter_setting.id}\/logo\.png/,
+        fit: [200, 40]
+      )
+      subject.render
+    end
```

Comment:
Testabdeckung hast du zwar zu 100%. Aber was mich eigentlich interessieren würde in diesem Test ist, welches Logo hat Vorrang wenn beide definiert sind? Wir hatten schon mal ein ganz ähnliches Problem mit Gruppen-Logos #839

Reply:
yes ... guter Punkt. habe noch einen test ergänz, verhält sich so wie erwartet :dancers: